### ISSUE TITLE
fine-tuning BERT, next sentence prediction loss is not decreasing

### ISSUE THREAD

**Issue by lazyfuzzypringle** [huggingface/pytorch-pretrained-BERT#649]
I run simple_lm_finetuning and monitor the loss change of next_sentence_loss and masked_lm_loss. The loss of masked token can converge but the next_sentence_loss is not decreasing.

For now I tried: tuning learning rate, change to optimizers from pytorch.optim, I checked the input_ids and the input looks good... I also ignore the loss from masked tokens and specifically train next sentence prediction but it didn't work.

 Did anyone face the same problem? Or where should I check to make it work? Thanks.

**Comment by mayankchatteron1** [comment]
same not working for me. NSP loss is not converging even though MLM loss is converging.
